src/Microsoft.AspNetCore.Mvc.Razor/Internal/DependencyContextRazorViewEngineOptionsSetup.cs

```diff
@@ -80,7 +80,7 @@ private static void SetCompilationOptions(RazorViewEngineOptions options, Extens
         {
             var parseOptions = options.ParseOptions;
             parseOptions = parseOptions.WithPreprocessorSymbols(
-                parseOptions.PreprocessorSymbolNames.Concat(compilationOptions.Defines));
+                parseOptions.PreprocessorSymbolNames.Concat(compilationOptions.Defines ?? Enumerable.Empty<string>()));
```

Comment:
`compilationOptions.Defines` shouldn't be null, that's a bug I need to fix if it is